codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/ServerOperationRegistryGeneration.kt

```diff
@@ -0,0 +1,281 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+package software.amazon.smithy.rust.codegen.server.smithy.generators
+
+import software.amazon.smithy.model.shapes.OperationShape
+import software.amazon.smithy.rust.codegen.rustlang.Attribute
+import software.amazon.smithy.rust.codegen.rustlang.CargoDependency
+import software.amazon.smithy.rust.codegen.rustlang.RustWriter
+import software.amazon.smithy.rust.codegen.rustlang.asType
+import software.amazon.smithy.rust.codegen.rustlang.rust
+import software.amazon.smithy.rust.codegen.rustlang.rustBlock
+import software.amazon.smithy.rust.codegen.rustlang.rustBlockTemplate
+import software.amazon.smithy.rust.codegen.rustlang.rustTemplate
+import software.amazon.smithy.rust.codegen.server.smithy.ServerRuntimeType
+import software.amazon.smithy.rust.codegen.smithy.CodegenContext
+import software.amazon.smithy.rust.codegen.smithy.RuntimeType
+import software.amazon.smithy.rust.codegen.smithy.protocols.HttpBindingResolver
+import software.amazon.smithy.rust.codegen.util.dq
+import software.amazon.smithy.rust.codegen.util.toSnakeCase
+
+/**
+ * ServerOperationRegistryGenerator
+ */
+class ServerOperationRegistryGenerator(
+    codegenContext: CodegenContext,
+    private val httpBindingResolver: HttpBindingResolver,
+    private val operations: List<OperationShape>,
+) {
+    private val serverCrate = "aws_smithy_http_server"
+    private val service = codegenContext.serviceShape
+    private val model = codegenContext.model
+    private val symbolProvider = codegenContext.symbolProvider
+    private val operationNames = operations.map { symbolProvider.toSymbol(it).name.toSnakeCase() }
+    private val runtimeConfig = codegenContext.runtimeConfig
+    private val codegenScope = arrayOf(
+        "Router" to ServerRuntimeType.Router(runtimeConfig),
+        "SmithyHttpServer" to CargoDependency.SmithyHttpServer(runtimeConfig).asType(),
+        "Phantom" to ServerRuntimeType.Phantom,
+        "StdError" to RuntimeType.StdError
+    )
+    private val operationRegistryName = "${service.getContextualName(service)}OperationRegistry"
```

Comment:
Unrelated to this PR.

I'm thinking that, given that the operation registry is part of the generated crate, perhaps it is too redundant to have it prefixed with the service name, given that it will appear in the namespace the user imports. So, perhaps calling it `OperationRegistry` is simpler and enough. Thoughts?

Reply:
I think you are right. Even if there are multiple services in the same model (is that even valid) we have to build a single Router so we can just use `OperationRegistry` and `OperationRegistryBuilder`. I am going to change this.

Reply:
Done.